### PR TITLE
Test/dec 20 manisha

### PR DESCRIPTION
**Pull request by manii283**

### CODE REVIEW

**Comment by manii283** [comment]
insert data, update data, view data is done .
delete is remain

**Comment by manii283** [comment]
country id and delete are done.
add new data to show session message is remain.

File: app/Http/Controllers/UniversityController.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+use App\Models\Country;
+use App\Models\University;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Session;
+
+class UniversityController extends Controller
+{
+    public function index()
+    {
+       $University = University::get();
```
Comment (ithqindia) [inline app/Http/Controllers/UniversityController.php:15]: Follow naming convention this should be $universities

File: app/Http/Controllers/UniversityController.php [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+       return view('/university', compact('University'));
```
Comment (ithqindia) [inline app/Http/Controllers/UniversityController.php:16]: view name should also be universities

File: app/Http/Controllers/UniversityController.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        ]);
+
+        $university = University::get();
+        return view('/show', compact('university'));
+    }
+        public function show(Request $request)
+    {
+        $userInformation = University::where('id', $request->id)->get()->first();
+        $countries = Country::get();
+        $user = [];
```
Comment (ithqindia) [inline app/Http/Controllers/UniversityController.php:49]: remove this line

File: app/Models/University.php [hunk 4]
```diff
@@ -9,7 +9,7 @@
 class University extends Model
 {
     use HasFactory;
-
+    protected $table = "universities";
```
Comment (ithqindia) [inline app/Models/University.php:12]: remove this lines as laravel automatically gets table name based on model name

File: routes/web.php [hunk 5]
```diff
@@ -170,3 +172,28 @@
...
+    $university =University ::get()->sortBy('name');
+    return view('show', compact('university'));
+});
+
+
+//user id
+Route::get('/university',function(){
+    $countries=Country::get()->sortBy('name');
+    return view('university',compact('countries'));
+});
```
Comment (ithqindia) [inline routes/web.php:199]: This is handled in Route::get('/university', [UniversityController::class, 'index']);
remove these lines

**Review by ithqindia — CHANGES_REQUESTED** [review]
Format all files

File: app/Http/Controllers/UniversityController.php [hunk 6]
```diff
@@ -39,39 +33,49 @@ public function store(Request $request)
...
-            $userInformation->is_enabled = (bool)$request->get('is_enabled');
-            $userInformation->save();
-            return view('success');
-        }
-        public function view(Request $request)
+    public function update(Request $request)
     {
-        $university = University::find($request->id);
-        return view('university-info', compact('university'));
+        $userInformation = University::find($request->id);
```
Comment (ithqindia) [inline app/Http/Controllers/UniversityController.php:51]: $userInformation  change $university

File: resources/views/show.blade.php [hunk 7]
```diff
@@ -1,43 +1,49 @@
...
+        <table class="table table-striped gy-7 gs-7">
+            <tr class="fw-bold fs-6 text-gray-800 border-bottom border-gray-200">
+                <th>#</th>
+                <th>Name</th>
+                <th>enabled</th>
+                <th>Action</th>
+            </tr>
+            @php
+                $counter = 1;
+            @endphp
```
Comment (ithqindia) [inline resources/views/show.blade.php:32]: Wrong.

File: resources/views/show.blade.php [hunk 8]
```diff
@@ -1,43 +1,49 @@
...
+            @foreach ($universities as $university)
+                <tr>
+                    <td>{{ $counter }}</td>
```
Comment (ithqindia) [inline resources/views/show.blade.php:35]: This should be $loop->index	https://laravel.com/docs/8.x/blade#the-loop-variable

File: routes/web.php [hunk 9]
```diff
@@ -179,21 +179,22 @@
...
+Route::get('/universities/{id}/view', [UniversityController::class, 'view']);
+Route::delete('/universities/{id}', [UniversityController::class, 'destroy']);
 
-    Route::get('/show', function () {
-    $university =University ::get()->sortBy('name');
-    return view('show', compact('university'));
+Route::get('/show', function () {
+    $universities = University::get()->sortBy('name');
+    return view('show', compact('universities'));
 });
```
Comment (ithqindia) [inline routes/web.php:193]: Update as per https://ithqworkspace.slack.com/archives/C02NSD7L98W/p1640319747003300

**Review by ithqindia — CHANGES_REQUESTED** [review]

**Comment by manii283** [comment]
code is updated but showing some errors in file

File: routes/web.php [hunk 10]
```diff
@@ -180,21 +180,21 @@
...
-Route::delete('/universities/{id}', [UniversityController::class, 'destroy']);
+Route::get('/universities/{id}/edit', [UniversityController::class, 'edit']);
+Route::put('/universities/{id}', [UniversityController::class, 'update']);
+// Route::get('/universities/{id}/view', [UniversityController::class, 'view']);
+Route::delete('/universities/{id}/', [UniversityController::class, 'destroy']);
 
-Route::get('/show', function () {
-    $universities = University::get()->sortBy('name');
-    return view('show', compact('universities'));
+Route::get('/universities', function () {
```
Comment (ithqindia) [inline routes/web.php:191]: This route is no longer required

File: routes/web.php [hunk 11]
```diff
@@ -180,21 +180,21 @@
...
+    $university = University::get()->sortBy('name');
+    return view('universities', compact('university'));
 });
 
-
 //user id
-Route::get('/universities', function () {
+Route::get('/universities-form', function () {
```
Comment (ithqindia) [inline routes/web.php:197]: This route is no longer required

**Review by ithqindia — CHANGES_REQUESTED** [review]

File: resources/views/includes/sidebar/menu.blade.php [hunk 12]
```diff
@@ -86,6 +86,12 @@
                     <span class="menu-title">Users</span>
                 </a>
             </div>
+            <div class="menu-item">
+                <a class="menu-link" href="/universities">
+                    <span class="menu-title">University table</span>
+                </a>
+            </div>
+            </div>
```
Comment (ithqindia) [inline resources/views/includes/sidebar/menu.blade.php:94]: Extra </div> remove

File: routes/web.php [hunk 13]
```diff
@@ -170,3 +172,18 @@
...
+Route::get('/universities/{id}', [UniversityController::class, 'show']);
+Route::get('/universities/{id}/edit', [UniversityController::class, 'edit']);
+Route::put('/universities/{id}', [UniversityController::class, 'update']);
+Route::delete('/universities/{id}', [UniversityController::class, 'destroy']);
+
+//user id
+Route::get('/universities-form', function () {
+    $countries = Country::get()->sortBy('name');
+    return view('/universities-form', compact('countries'));
+});
```
Comment (ithqindia) [inline routes/web.php:189]: Delete these lines

**Review by ithqindia — CHANGES_REQUESTED** [review]

**Comment by ithqindia** [comment]
Merged locally

**Pull request closed without merging**